clients/cobol-lsp-vscode-extension/src/services/settings/CopybookResolver.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+/**
+ * This interface represent the base class that define the API to resolve a JSON segment or a list of paths
+ * defined by the user in the settings configuration in a list of URI Paths available on the filesystem.
+ */
+export interface CopybookResolver {
```

Comment:
My idea is to define the resolutions of a copybook as different strategy and the interface is a good design pattern to made available to the client rather than the concrete class

Reply:
Make sense. Maybe it's too early to apply a strategy pattern because we don't have that much logic for now. But it's already here, then ok )

Reply:
yeye agree, but as u said I just prepared the field ^__^